bin/beaver-install

```diff
@@ -24,13 +26,21 @@ do
     esac
 done
 
-if test -r "Dockerfile${DIST:+.$DIST}"
-then
-    # Build the docker image using .$DIST as suffix if present
-    beaver docker build -f "Dockerfile${DIST:+.$DIST}" "$@" .
-else
-    # Build the docker image based on a beaver.Dockerfile base file
-    beaver docker gen-dockerfile -i beaver.Dockerfile \
+if [ "$#" -ne 0 ] && [ "$1" = "--" ]; then shift; fi
+
+generate_and_build()
+{
+    if test -r "Dockerfile${DIST:+.$DIST}"
+    then
+        # Build the docker image using .$DIST as suffix if present
+        beaver docker build -f "Dockerfile${DIST:+.$DIST}" "$@" .
+    else
+        # Build the docker image based on a beaver.Dockerfile base file
+        beaver docker gen-dockerfile \
+            -i beaver.Dockerfile \
             -o beaver.Dockerfile.generated
-    beaver docker build -f beaver.Dockerfile.generated "$@" .
-fi
+        beaver docker build -f beaver.Dockerfile.generated "$@" .
+    fi
+}
+
+(cd $BEAVER_CONTEXT_DIR && generate_and_build $@)
```

Comment:
I don't like this. The idea was to keep the top-level project directory as the working directory when invoking `docker build`, and to copy the `.dockerignore` file in the "context" directory to the top-level. This way the `Dockerfile` writer can have full control over what's sent to the docker daemon on build-time, and can be anything in the project. Limiting users to files located in `docker/` have proven to be problematic in some cases in the past, I wouldn't want to lose the flexibility here.

Reply:
I don't understand why does it matter. It would surely matter a lot if we used "idiomatic" version but with beaver you must have some shell script to bind stages together no matter what. Copying/linking required files into context is just one extra step in such script - and keeps amount of magic involved in the process under control.

Reply:
(with proposed approach `.dockerignore` files are not really relevant anymore indeed)

Reply:
Ping @leandro-lucarella-sociomantic interested in hearing how you see this